kvlds/btree_sync.c

```diff
@@ -223,9 +223,10 @@ btree_sync(struct btree * T, int (* callback)(void *), void * cookie)
 
 	/* Sanity check the number of pages serialized. */
 	assert(pn == npages);
+	assert(npages <= UINT32_MAX);
```

Comment:
It's a bit more than this -- the serialized request to LBS, which is npages * pagelen + overhead bytes, needs to fit into a 32-bit kivaloo wire packet.  We should probably assert that npages * pagelen isn't too big.

I have code somewhere which makes sure that we don't handle too many requests at once -- the idea was to make sure that we wouldn't need to load too many pages into memory, but it should also work for making sure that we don't dirty too many pages at once.

Reply:
I'm always happy to have explicit checks, but what you've described is already done implicitly.  Namely,

- `proto_lbs_request_append_blks()` is passed `npages`.
- `lib/proto_lbs/proto_lbs_client.c` line 306 (the code for `proto_lbs_request_append_blks()`) calculates

      size_t len = 16 + nblks * blklen;

  which is then passed to `wire_requestqueue_add_getbuf()`.

- in turn, that's passed to `wire_writepacket_getbuf()` in `lib/wire/wire_writepacket.c` line 24, which ensures that `len <= UINT32_MAX` and `len <= SIZE_MAX - 20`.

Reply:
Yes, there's an assertion -- but we should be asserting things which should never be false.  If it's possible for an assertion to fail, it means there's a bug somewhere else.

In this case, we should make sure that we never end up with so many dirty pages that we would *want* to write too much at once.